R/onLoad.R

```diff
@@ -66,6 +66,15 @@
       lockBinding("rbind.data.frame",baseenv())
     }
   }
+  if (session_r_version < "3.6.0") {
+    # no delayed registration support for NAMESPACE; perform it manually
+    if (isNamespaceLoaded("knitr")) {
```

Comment:
this is a really nice defensive tactic for avoiding loading the `Enhances` namespace if we can 💪

Reply:
IINM we don't need to do the `setHook()` step when `isNamespaceLoaded()`, so your original code was a no-op in that case; I put `setHook()` inside a branch for some clarity/to reduce potential confusion.

Reply:
Good point. So not quite a no-op, even if it'll only show up in extremely odd use cases.